File: codegen/results/v001/qwen3.5_4b-nvfp4_nothink/20260409_092330/python/004_json_filter/solution.py
import json

# Read the input file
with open('input/data.json', 'r') as f:
    data = json.load(f)

# Filter records where active is True and age >= 30, then sort by name
filtered_data = [
    record for record in data 
    if record['active'] is True and record['age'] >= 30
]
filtered_data.sort(key=lambda x: x['name'])

# Output the result as JSON
print(json.dumps(filtered_data))

File: codegen/results/v001/qwen3.5_4b-nvfp4_nothink/20260409_092330/typescript/004_json_filter/solution.ts
import { readFileSync } from 'fs';

const dataPath = process.argv[2];

if (!dataPath) {
  console.error('Usage: node script.js <input_file>');
  process.exit(1);
}

try {
  const rawData = JSON.parse(readFileSync(dataPath, 'utf8'));
  
  if (!Array.isArray(rawData)) {
    console.log('[]');
    process.exit(0);
  }

  const result = rawData
    .filter(record => record.active === true && record.age >= 30)
    .sort((a, b) => a.name.localeCompare(b.name));

  console.log(JSON.stringify(result, null, 2));
} catch (e) {
  console.error('Error:', e.message);
  process.exit(1);
}